libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/mw300_rd/NetworkInterface.c

```diff
@@ -148,22 +148,16 @@ void handle_data_packet(const t_u8 interface, const t_u8 *rcvdata,
 
 BaseType_t xNetworkInterfaceInitialise( void )
 {
-    static BaseType_t xMACAdrInitialized = pdFALSE;
     uint8_t ret;
-
-    if (xInterfaceState == INTERFACE_UP) {
-	if (xMACAdrInitialized == pdFALSE) {
-	    mac_addr_t mac_addr;
-	    ret = wifi_get_device_mac_addr(&mac_addr);
-	    if (ret != WM_SUCCESS) {
+    mac_addr_t mac_addr;
+    
+	ret = wifi_get_device_mac_addr(&mac_addr);
+	if (ret != WM_SUCCESS) {
 		net_d("Failed to get mac address");
 		return pdFALSE;
-	    }
-	    FreeRTOS_UpdateMACAddress(mac_addr.mac);
-	    xMACAdrInitialized = pdTRUE;
-	 }
-	 return pdTRUE;
-    }
+	}
+	FreeRTOS_UpdateMACAddress(mac_addr.mac);
+
     return pdFALSE;
```

Comment:
Seems the function always returns pdFALSE now. Should the return type checked by TCP IP stack to know if the interface is initialized.

Reply:
Good catch.. was too deep into the fix, left the return out... 

(The return logic followed the same logic we previously have. I think, the next step is reaching out to Marvell to see if there are good this this fix. )